### PR TITLE
Insertions accumulating in AbstractLocusIterator bugfix

### PR DESCRIPTION
**Pull request by nkarulin**
The current version of `AbstractLocusIterator` class works the following way: when new read comes, it completes all accumulated loci with position to the left of the read start. When the flag `includeIndels` is specified, the read starting with insertions won't complete the locus at the position to the left of the read start. So, iterator will keep previous locus, which will store insertion info. 
The bug appears when a read that does not start with insertion(REC1) comes before the read that does start with insertion (REC2) at the same position:
```
REF                     :  ATGC....TGCATGCATGC
REC (Pos=5, CIGAR: 3M)  :          TGC
REC (Pos=5, CIGAR: 3I3M):      TGCATGC
```
In this case the first read without insertion will complete all loci at the previous positions. As a result there is no locus in which we should have accumulated the insertion from the second read which leads to `IllegalStateException`.
We propose to store one more locus in accumulator before completion to make sure that all records (including those that start with insertion) on current position were accumulated.

### CODE REVIEW

**Comment by yfarjoun** [comment]
Could you please include a testcase that covered the bug you are fixing?

**Comment by yfarjoun** [comment]
this is fine. Thanks.

**Comment by lbergelson** [comment]
@yfarjoun Did you do a review pass on this or just check for tests?

**Comment by yfarjoun** [comment]
I just checked for tests.

**Comment by yfarjoun** [comment]
also @kvg did you find time to take a look?

File: src/test/java/htsjdk/samtools/util/SamLocusIteratorTest.java [hunk 1]
```diff
@@ -263,6 +263,8 @@ public void testStartWithInsertion() {
         final SAMRecordSetBuilder builder = getRecordBuilder();
         // add records up to coverage for the test in that position
         final int startPosition = 165;
+        // add record without insertions first
+        builder.addFrag("record", 0, startPosition, true, false, "33M", null, 10);
```
[the remark this answers is not in the document]
Comment (kvg) [reply src/test/java/htsjdk/samtools/util/SamLocusIteratorTest.java]: Agreed - it's a super unusual case that I wouldn't really anticipate being a problem, but technically there should be a test to cover it.  I think it'll be sufficient for now to accept these changes as is and add a test ourselves covering this condition.

**Comment by lbergelson** [comment]
@yfarjoun @kvg This change causes `FastWgsMetricsCollectorTest.testForComplicatedCigar()` to fail with 

```
java.lang.AssertionError: Excluded by overlap: expected [13] but found [11]
Expected :13
Actual   :11
at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
	at org.testng.Assert.assertEquals(Assert.java:116)
	at org.testng.Assert.assertEquals(Assert.java:284)
	at picard.analysis.FastWgsMetricsCollectorTest.testForComplicatedCigar(FastWgsMetricsCollectorTest.java:213)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:643)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:820)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1128)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:73)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)
```
Is this working as intended?

**Comment by lbergelson** [comment]
I should have specified this is in picard... not a test in htsjdk.

**Pull request merged**